core/dbt/include/global_project/macros/materializations/models/materialized_view/materialized_view.sql

```diff
@@ -0,0 +1,71 @@
+{%- materialization materialized_view, default -%}
+  {% set full_refresh_mode = should_full_refresh() %}
+  {%- set existing_relation = load_cached_relation(this) -%}
+  {%- set target_relation = this.incorporate(type='view') -%}
+  {%- set intermediate_relation =  make_intermediate_relation(target_relation) -%}
+
+  -- the intermediate_relation should not already exist in the database; get_relation
+  -- will return None in that case. Otherwise, we get a relation that we can drop
+  -- later, before we try to use this name for the current operation
+  {%- set preexisting_intermediate_relation = load_cached_relation(intermediate_relation) -%}
+  /*
+     This relation (probably) doesn't exist yet. If it does exist, it's a leftover from
+     a previous run, and we're going to try to drop it immediately. At the end of this
+     materialization, we're going to rename the "existing_relation" to this identifier,
+     and then we're going to drop it. In order to make sure we run the correct one of:
+       - drop view ...
+       - drop table ...
+     We need to set the type of this relation to be the type of the existing_relation, if it exists,
+     or else "view" as a sane default if it does not. Note that if the existing_relation does not
+     exist, then there is nothing to move out of the way and subsequentally drop. In that case,
+     this relation will be effectively unused.
+  */
+  {%- set backup_relation_type = 'view' if existing_relation is none else existing_relation.type -%}
+  {%- set backup_relation = make_backup_relation(target_relation, backup_relation_type) -%}
+  -- as above, the backup_relation should not already exist
+  {%- set preexisting_backup_relation = load_cached_relation(backup_relation) -%}
+  -- grab current tables grants config for comparision later on
+  {% set grant_config = config.get('grants') %}
+
+  {{ run_hooks(pre_hooks, inside_transaction=False) }}
+
+  -- drop the temp relations if they exist already in the database
+  {{ drop_relation_if_exists(preexisting_intermediate_relation) }}
+  {{ drop_relation_if_exists(preexisting_backup_relation) }}
+
+  -- `BEGIN` happens here:
+  {{ run_hooks(pre_hooks, inside_transaction=True) }}
+
+  -- cleanup
+  -- move the existing view out of the way
+  {% if existing_relation is none %}
+      {% set build_sql = strategy__materialized_view__create(target_relation, sql) %}
+  {% elif full_refresh_mode or existing_relation.type != 'view' %}
+      {% set build_sql = strategy__materialized_view__full__refresh(target_relation, backup_relation, sql) %}
+  {% else %}
+      {% set build_sql = strategy__materialized_view__full__refresh(target_relation) %}
+  {% endif %}
+
+    -- possible conditional wrapper for build_sql statement based on configs as a option?
+    -- build model
+  {% call statement("main") %}
```

Comment:
We need to make sure this can actually exist outside of the if block above. For example, does `adapter.rename()` occur when you call it? Or does it just return the sql to do so? We're fine if it's the latter. But if it's the former, we'll do all the thing that happen immediately during the if block, and then do all the things that we return sql for during this phase. That may not be the right order.